xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodecSurface.cpp

```diff
@@ -156,6 +156,17 @@ bool CRendererMediaCodecSurface::RenderUpdateVideoHook(bool clear, DWORD flags,
     mci->RenderUpdate(srcRect, dstRect);
   }
 
+  /** @todo This is a really bad hack and not meant to be
+  * a solution. This needs proper fixing when android maintainer
+  * is back. Real solution: Block until your next vblank.
```

Comment:
Yes, you are right. Then what Pi / IMX want to do is not enough, as we need the interval, e.g. the clock to decided "how much" already has passed.